emloop/api.py

```diff
@@ -0,0 +1,85 @@
+from typing import Union, Optional, CommentedMap
+import copy
+import os
+
+from .utils.yaml import yaml_to_str
+from .utils.config import load_yaml
+from .utils.reflection import parse_fully_qualified_name, create_object
+from .cli.util import find_config
+
+
+def load_yaml(path: str):
+# returns contents of YAML file as CommentedMap
+
+
+def create_output_dir(config: CommentedMap):
+# creates output directory and returns path to it
+# name of the output dir is taken from config["model"]["name"]
+# writes config to output dir
+
+
+def create_model(config: CommentedMap, output_dir=None, dataset=None):
+
+
+def load_model(restore_path: str, dataset=None, output_dir=None):
+
+
+def create_dataset(config: CommentedMap, output_dir: Optional[str]=None):
+
+
+def create_hooks(config: CommentedMap, model=None, dataset=None, output_dir=None):
+# creates hooks and returns them as list
+
+
+def create_all(config: CommentedMap):
+# creates output_dir, dataset, model and hooks
+    output_dir = create_output_dir(config)
+    dataset = create_dataset(config, output_dir=output_dir)
+    model = create_model(config, dataset=dataset, output_dir=output_dir)
+    hooks = create_hooks(config, model=model, dataset=dataset, output_dir=output_dir)
+
+    return output_dir, dataset, model, hooks
+
+
+def load_all(restore_path: str):
+# loads model, creates new output_dir, hooks and  dataset according to config in restore_path
+
+
+def start(hooks):
```

Comment:
I would rather have `start` and `finish` as a resource block (i.e., `with emloop.session(...) as sess`).